flex/engines/graph_db/app/builtin/shortest_path_among_three.cc

```diff
@@ -0,0 +1,231 @@
+    return false;
+  }
+
+  std::vector<int64_t> v1v2result_;
+  std::vector<int64_t> v2v3result_;
+  std::vector<int64_t> v1v3result_;
+
+  bool find_flag = true;
+  if (!ShortestPath(txn, label_v1, index_v1, label_v2, index_v2, v1v2result_)) {
+    find_flag = false;
+  }
+  if (find_flag &&
+      !ShortestPath(txn, label_v2, index_v2, label_v3, index_v3, v2v3result_)) {
+    find_flag = false;
+  }
+  if (find_flag &&
+      !ShortestPath(txn, label_v1, index_v1, label_v3, index_v3, v1v3result_)) {
+    find_flag = false;
+  }
+  std::string result_path = "";
+  if (find_flag) {
+    // connect two shortest paths
+    std::vector<int64_t> TSP =
+        ConnectPath(v1v2result_, v2v3result_, v1v3result_);
+    for (auto it = TSP.begin(); it != TSP.end(); ++it) {
+      if (std::next(it) != TSP.end()) {
+        result_path += std::to_string(*it) + "--";
+      } else {
+        result_path += std::to_string(*it);
+      }
+    }
+  } else {
+    result_path = "no path find!";
+  }
+
+  // create result string
+  results::CollectiveResults results;
+  auto result = results.add_results();
+  result->mutable_record()
+      ->add_columns()
+      ->mutable_entry()
+      ->mutable_element()
+      ->mutable_object()
+      ->set_str(result_path);
+
+  output.put_string_view(results.SerializeAsString());
+  txn.Commit();
+  return true;
+}
+
+bool ShortestPathAmongThree::ShortestPath(const gs::ReadTransaction& txn,
+                                          label_t v1_l, vid_t v1_index,
+                                          label_t v2_l, vid_t v2_index,
+                                          std::vector<int64_t>& result_) {
+  Schema schema_ = txn.schema();
+  int vertex_size_ = (int) schema_.vertex_label_num();
+  int edge_size_ = (int) schema_.edge_label_num();
+
+  std::unordered_map<int64_t, int64_t> parent;
+  std::vector<label_t> nei_label_;
+  std::vector<vid_t> nei_index_;
+  int64_t v1_id = txn.GetVertexId(v1_l, v1_index).AsInt64();
+  int64_t v2_id = txn.GetVertexId(v2_l, v2_index).AsInt64();
+
+  parent[v1_id] = -1;
+  nei_label_.push_back(v1_l);
+  nei_index_.push_back(v1_index);
+  std::unordered_set<int64_t> visit;
+  visit.insert(v1_id);
+
+  std::vector<label_t> next_nei_labels_;
+  std::vector<vid_t> next_nei_indexs_;
+  bool find = false;
+  while (!nei_label_.empty() && !find) {
+    for (long unsigned int i = 0; i < nei_index_.size(); i++) {
+      for (int j = 0; j < vertex_size_; j++) {
+        for (int k = 0; k < edge_size_; k++) {
+          if (schema_.has_edge_label(label_t(nei_label_[i]), label_t(j),
+                                     label_t(k))) {
+            auto outedges = txn.GetOutEdgeIterator(
+                nei_label_[i], nei_index_[i], j,
+                k);  // 1.self_label 2.self_index 3.edge_label 4.nei_label
+            while (outedges.IsValid()) {
+              auto neighbor = outedges.GetNeighbor();
+              int64_t v_id = txn.GetVertexId(j, neighbor).AsInt64();
```

Comment:
Is this conversion necessary here? maybe we only need oid when generating the result.

Reply:
This is necessary because currently only the vid of int64 type is adapted, and for other types of vid, it seems that the C++ interfaces are still not perfect at present.

Reply:
Are there vertices with different labels? We can obtain the oid by using the GetVertexId interface, but the oid is not unique within a graph. The combination of (label, vid) can uniquely define a vertex.

Reply:
If there is only one label, then the vid is sufficient to uniquely define a vertex, so we do not need to perform this conversion by calling `GetVertexId`.

Reply:
Thank you for your suggestion. The latest commit has been modified to use label and vid as unique identifiers.
There is a better user experience when returning the results, which are displayed as oid. The demo is shown in the following figure

Reply:
![5bbfb5d7416319bec7e5474c57a90ea](https://github.com/user-attachments/assets/47e79f3f-ee62-4334-9b7e-e0522aaebd53)
![d0b63729f3fc2d8a18f3db71e26aba5](https://github.com/user-attachments/assets/e79cffd2-05ef-4452-b6a4-99eed673ddf5)
![96ea63e2ff7b71e93a947e1dbe16ccd](https://github.com/user-attachments/assets/9527c221-23f7-4316-a603-0ca01597cb4e)